js/customize-concurrency.js

```diff
@@ -0,0 +1,70 @@
+/* eslint-disable no-extra-parens */
+
+( function( api ) {
+	'use strict';
+
+	var component;
+
+	if ( ! api.Concurrency ) {
+		api.Concurrency = {};
+	}
+
+	component = api.Concurrency;
+
+	component.data = {};
+
+	/**
+	 * Inject the functionality.
+	 *
+	 * @return {void}
+	 */
+	component.init = function() {
+		api.bind( 'ready', function() {
+			api.each( function( setting ) {
+				setting['concurrency_timestamp'] = _customizeConcurrency.session_start_timestamp;
+			} );
+
+			api.bind( 'add', function( setting ) {
+				// todo get the timestamp into the ajax response when additional settings are loaded so that we can use it here
+				setting['concurrency_timestamp'] = _customizeConcurrency.session_start_timestamp;
+			} );
+
+			api.bind( 'saved', function( data ) {
+				wp.customize.each( function( setting ) {
+					if ( data.saved_post_setting_values.hasOwnProperty( setting.ID ) ) {
+						setting['concurrency_timestamp'] = data.concurrency_session_timestamp;
+					}
+				} );
```

Comment:
@lgedeon here you'll also want to make sure you update whatever you have stored as the `session_start_timestamp` to be now `data.concurrency_session_timestamp`, so that every time you save the initial timestamp moves forward.